fix: spacing bugs

jnm2377 opened the pull request:
Fixes spacing issues in image clusters (resulting from improper use of rows), and overrides Gatsby vertical spacing issues in colors page and SwatchPalette styles.

jnm2377 commented:
cc: @mjabbink @sadek @rcwakelin Not sure who should be reviewing this.

specifically fixes:
- bugs seen in the ColorGrid & SwatchPalette in colors page: https://github.ibm.com/brand/design-language/issues/67
- image cluster spacing in IBM Logos/8-bar page: https://github.ibm.com/brand/design-language/issues/68#issuecomment-15854450
- image cluster spacing in elements/2x-grid page: https://github.ibm.com/brand/design-language/issues/68#issuecomment-15854837
- image cluster spacing in photography/overview page
- image cluster spacing in colors page

vpicone commented on src/components/SwatchPaletteWidget/_swatch-palette-widget.scss:
```diff
@@ -90,3 +90,7 @@
   }
 }
+
+[class*='--col'].#{$prefix}--swatch-palettes > * + * {
```
If a 'stack' doesn't have different types of components, there's no need for the loose selector, we can just define `--space` with the child.

```suggestion
.#{$prefix}--swatch-palette {
```

vpicone commented:
Does this one have issues to go along with it?

jnm2377 commented:
@vpicone the issues that were opened were all in the enterprise repo.

jnm2377 commented:
Rel: #349

jnm2377 commented:
@rc-wakelin @mjabbink what designer should be reviewing this PR?

rcwakelin commented:
@sadek can you take review this ticket?

jnm2377 commented:
@sadekbazaraa could you review this?

sadekbazaraa commented:
@jnm2377 I still see various spacing above captions as I scroll down multiple pages
https://deploy-preview-350--design-language-website.netlify.com/design/language/elements/2x-grid
https://deploy-preview-350--design-language-website.netlify.com/design/language/elements/ibm-logos/8-bar

– The spacing on the color palette page seems fixed
– Photography clusters seem good in general

I noticed this cluster on the 8-bar page that is off

![Screen Shot 2019-11-22 at 4 23 53 PM](https://user-images.githubusercontent.com/33162280/69461505-b1ca8c00-0d44-11ea-9f93-ad6b7feaf143.png)

mjabbink commented:
Spacing to tight @sadek and @jeanservaas can confirm spacing but it needs to be more space.

<img width="1308" alt="Screen Shot 2019-11-27 at 9 57 42 PM" src="https://user-images.githubusercontent.com/32881239/69773514-14ff5880-1161-11ea-8838-5ad00bd4f0fa.png">

sadekbazaraa commented:
@jnm2377 @mjabbink the original design had 64px of space after the video. Under the new spacing scheme, I believe the spacing has been bumped up to 80px which would be similar to the 
spacing after other videos such as:
https://deploy-preview-350--design-language-website.netlify.com/design/language/elements/2x-grid

![Screen Shot 2019-12-02 at 11 00 07 AM](https://user-images.githubusercontent.com/33162280/69974692-c43a8780-14f3-11ea-9133-b81b496e4a7d.png)

This page also has a bit of a unique pattern relative to all the other pages so I'd be fine with a custom override or additional spacer in this case.

jnm2377 commented:
@sadekbazaraa @mjabbink I some of the spacing issues you pointed out. I _think_ I fixed everything, but I really could only fix obvious image/caption issues. Some other issues may not appear wrong to me because I don't know the specs of all these pages. Without specific issues with expected specs for a specific page, I have no way of knowing what is a visual bug.

For review: the pages I updated were:

- app-icons/design
- icons/design
- philosophy/pov
- ibm-logos/* (all of them)

mjabbink commented:
Is there a preview link to look over?

jnm2377 commented:
@mjabbink https://deploy-preview-350--design-language-website.netlify.com/

Btw you'll  always find the latest preview links at the bottom of the page under the `checks` accordion item. For IDL it's the Netlify deploy preview, but I believe Carbon and Gatsby have a different tool they use for deploy previews (still under checks though).

<img width="843" alt="Screen Shot 2019-12-04 at 9 43 32 AM" src="https://user-images.githubusercontent.com/32556167/70157106-90d63500-167a-11ea-9b88-a2f2b66c33bc.png">

mjabbink commented:
In Philosophy page the space between the paragraph is now too large. It should just be a full return in the text field. Just like here:https://www.ibm.com/design/language/philosophy/pov


<img width="799" alt="Screen Shot 2019-12-04 at 10 16 37 AM" src="https://user-images.githubusercontent.com/32881239/70159854-2e336800-167f-11ea-8418-79f24d32c1ea.png">

mjabbink commented:
the spacing above caption should be 16px. @jeanservaas can you help with this. 

current in preview link is too large:
<img width="1009" alt="Screen Shot 2019-12-04 at 10 20 11 AM" src="https://user-images.githubusercontent.com/32881239/70160233-c893ab80-167f-11ea-8f77-e92ab2f88f76.png">

mjabbink commented:
@alisonjoseph @jnm2377 there are still major spacing issues. I recommend meeting and going over these in person. Capturing them in an issue would take too long.

jnm2377 commented:
@mjabbink I fixed the homepage, image and philosophy spacing issues. Still have to work on updating the mini image galleries to actually use the image gallery component. It might be a bit trickier than I anticipated or require a different design since we don't have any offset columns as part of the image gallery component. 

referring to this page:
<img width="983" alt="Screen Shot 2019-12-05 at 4 13 21 PM" src="https://user-images.githubusercontent.com/32556167/70278589-302e2180-177a-11ea-939b-a4f4dda8c143.png">

If we want this design specifically, it might require just overriding the current row spacing instead of using the image gallery component, or enhancing the image gallery component in gatsby, which might take a while.

I think it's worth merging this PR so the IDL site will at least have those major visual bugs fixed, since this is a minor bug. This PR contains the fix for all the colors pages and image clusters that didn't have margin, which has been on the current IDL site for weeks now..

jnm2377 commented:
cc: @sadekbazaraa see above comment as well.

sadekbazaraa commented:
@jnm2377 I agree that we should merge and then revisit the larger issues around having flexibility with the image gallery component. Let's not hold up because of the spacing here.

jnm2377 commented:
@mjabbink have you had a chance to look at this again? I added a temporary override to the photography page gallery rows so that they have 32px margin top. This should be good to go now. I think other issues can be captured in a different PR.

jnm2377 commented:
@mjabbink is this good to go? I'd like to get this PR merged before I'm OOO for the holidays (in 2 days).

mjabbink commented:
👍

mjabbink commented:
Let’s get this deployed!
